manifests/src/upload_index.sh

```diff
@@ -1,20 +1,59 @@
 # Find the latest version of the dataset
-ZENODO_ENDPOINT="https://zenodo.org"
-DEPOSITION_PREFIX="${ZENODO_ENDPOINT}/api/deposit/depositions"
 ORIGINAL_ID="13892061"
 FILE_TO_VERSION="manifests/profile_index.csv"
-FILENAME=$(echo ${FILE_TO_VERSION} | sed 's+.*/++g')
+METADATA_JSON='{
+  "metadata": {
+    "title": "The Joint Undertaking for Morphological Profiling (JUMP) Consortium Datasets Index",
+    "creators": [
+      {
+        "name": "The JUMP Cell Painting Consortium"
+      }
+    ],
+    "upload_type": "dataset",
+    "access_right": "open"
+  }
+}'
+
+ZENODO_ENDPOINT="https://zenodo.org"
+DEPOSITION_PREFIX="${ZENODO_ENDPOINT}/api/deposit/depositions"
+
+FILENAME=${FILE_TO_VERSION##*/}
 
 echo "Checking that S3 ETags match their local counterpart"
-S3_ETAGS=$(cat ${FILE_TO_VERSION} | tail -n +2 | cut -f2 -d',' | xargs -I {} -- curl -I --silent "{}" | grep ETag | awk '{print $2}' | sed 's/\r$//' | md5sum | cut -f1 -d" ")
-LOCAL_ETAGS=$(cat ${FILE_TO_VERSION} | tail -n +2 | cut -f3 -d',' | md5sum | cut -f1 -d" ")
 
-echo "Remote ${S3_ETAGS} vs Local ${LOCAL_ETAGS} values"
-if [ "${S3_ETAGS}" != "${LOCAL_ETAGS}" ]; then
+# Extract URLs and ETags
+url_column=$(head -n1 "${FILE_TO_VERSION}" | tr ',' '\n' | grep -n "url" | cut -d':' -f1)
+urls=$(awk -F',' -v col="${url_column}" 'NR>1 {gsub(/^"|"$/, "", $col); print $col}' "${FILE_TO_VERSION}")
+
+etag_column=$(head -n1 "${FILE_TO_VERSION}" | tr ',' '\n' | grep -n "etag" | cut -d':' -f1)
+local_etags=$(awk -F',' -v col="${etag_column}" 'NR>1 {gsub(/^"|"$/, "", $col); print $col}' "${FILE_TO_VERSION}")
+
+s3_etags=""
+while IFS= read -r url; do
+  etag=$(curl -I --silent "$url" | awk '/[eE][tT]ag:/ {print $2}' | tr -d '\r"')
+  s3_etags+="${etag}\n"
+done <<< "${urls}"
+
+# Remove the trailing newline from s3_etags
+s3_etags=$(echo -e "${s3_etags}" | sed '/^$/d')
+
+# Calculate checksums for comparison
+s3_etags_hash=$(echo -e "${s3_etags}" | md5sum | cut -f1 -d" ")
```

Comment:
Is the -e flag due to the etags having the '-' value? I suppose it covers a fringe case, but I'm not sure which one.